### PR TITLE
add data for CHAID tree regression

### PR DESCRIPTION
**Pull request by Hephaestus12**

### CODE REVIEW

**Comment by gf712** [comment]
Isn’t it odd that all weights are 1?

**Comment by Hephaestus12** [comment]
Hmm yes, I just saw this, why migt this be happening? And how can I debug?

This is the original meta-example PR #5065

**Comment by karlnapf** [comment]
It might just be a mismatch between dataset and algorithm, many of our examples are used on nonsensical data. There are a few examples for 1d or 2d regression/classification data with linear/nonlinear features. Try one of those.
Also run the old python example and check the output there

**Comment by Hephaestus12** [comment]
> It might just be a mismatch between dataset and algorithm, many of our examples are used on nonsensical data. There are a few examples for 1d or 2d regression/classification data with linear/nonlinear features. Try one of those.
> Also run the old python example and check the output there

The old python example actually ran on randomly generated data, not a specific dataset.

One thing to note is that the CARTtree regression, which runs on the same dataset produces the same output of all weights equal to 1, so I guess it is working fine?

https://github.com/shogun-toolbox/shogun-data/blob/d96c5c80d74793c4809920e33bdfe1b8648e5c9e/testsuite/meta/regression/cartree.dat#L1

**Comment by gf712** [comment]
Actually, this is the weight of each sample, not a learnt set of weights. It is probably not very useful to serialise this, as weights are always equals to 1 if they are not preset by the user. The prediction is pretty bad (have a look at the test set labels), but I think that might just be a combination of not enough data and weak signal. @karlnapf I don't know if it is worth having a larger dataset/more appropriate dataset just to have a a more reasonable prediction?

**Comment by vigsterkr** [comment]
it's just an integration test

**Comment by karlnapf** [comment]
It is also an example that users might look at. It doesnt need to be great, but I think it makes sense to at least make it nonsensical ...
We have super simple 1d linear datasets for regression where each algorithm should work ... I made them exactly for that. So I suggest to simply use that

**Comment by Hephaestus12** [comment]
What changes should I make to this?
Or the actual meta example:  [#5065](https://github.com/shogun-toolbox/shogun/pull/5065)

**Comment by karlnapf** [comment]
in the meta example load the linear regression data
E.g. [this](https://github.com/shogun-toolbox/shogun-data/blob/master/toy/regression_1d_linear_features_train.dat) is the training data. There is also test data. This *has* to work otherwise something is wrong

**Comment by karlnapf** [comment]
@gf712 re serializing the weights, I think that is needed. We could make it such that the weight vector is empty if the user hasnt set weights yet...minor refactor in the code. Could open as issue for that?

**Comment by gf712** [comment]
> @gf712 re serializing the weights, I think that is needed. We could make it such that the weight vector is empty if the user hasnt set weights yet...minor refactor in the code. Could open as issue for that?

I just meant for the integration test it doesn't make a lot of sense, i.e. in the meta example `RealVector weights = chaidtree.get_real_vector("weights")` doesn't really bring much. For the actual model serialisation I guess it is fine, but yes, could maybe think about having an empty vector in case there is no user defined weight vector.

> in the meta example load the linear regression data
E.g. this is the training data. There is also test data. This has to work otherwise something is wrong

I think that is the dataset being used here, I guess the predictions aren't as accurate as I thought they could be, but it's fine. Maybe some other parameter combination would be better...

**Comment by karlnapf** [comment]
Ah yes let’s just remove the weight extraction and merge then...

**Comment by geektoni** [comment]
Why do you also add weighted degree string data here? Is it supposed to be there?

**Comment by Hephaestus12** [comment]
No, I had made this PR before LiuYuhui merged his PR, so when I pulled his changes and applied git pull --rebase, this happened.

**Comment by Hephaestus12** [comment]
And also there were merge conflicts I was trying to resolve when I did this. Shouldn't it be there?

**Comment by geektoni** [comment]
> And also there were merge conflicts I was trying to resolve when I did this. Shouldn't it be there?

No, it shouldn't be there.

**Comment by Hephaestus12** [comment]
Okay, I'll manually delete it for now and push.

**Comment by geektoni** [comment]
Here you are deleting the `weighted_degree_string.dat` file :) remove the last commit you made (e.g., with `git reset --hard HEAD~1`) and then force push again. Then this PR should be ready.

**Comment by Hephaestus12** [comment]
This PR is fine now right? However, the other PR still says that there is a merge conflict in the data folder, will that get sorted after we merge this PR?

**Comment by karlnapf** [comment]
No it doesn’t sort itself out, you need to sort the other one as well. And because you need to force push eventually to do that you will need to update the submodule in here again as well

**Comment by karlnapf** [comment]
OK one last thing, could you please squash the commits here and force push again (we only want a single commit per PR in the data repo, and I cannot squash them here)...you will need to update the main PR then again as well with the new version

**Comment by Hephaestus12** [comment]
This is what you meant right?
I did a `git reset --soft HEAD~2` and then commited and pushed again.

**Comment by karlnapf** [comment]
yes, single commit now. Let's wait for CI :)

**Comment by karlnapf** [comment]
you need to update the develop PR as well now. Everytime the data PR changes you need to do that
